content/exchange/artifacts/Linux.Environment.Variables.yaml

```diff
@@ -0,0 +1,24 @@
+name: Linux.Environment.Variables
+author: Sikha Puthanveedu @SikhaMohan
+description: Get Environment Variables from Linux System. 
+             Collect from current shell session and the system-wide configuration file.
+parameters:
+  - name: envfile
+    default: /etc/environment
+precondition: |
+      SELECT OS From info() where OS = 'linux'
+sources:
+  - name: Session Environment Variables
+    query: |
+        SELECT parse_string_with_regex(regex="^([^=]+)", string=Stdout).g1 AS Variable,
+               parse_string_with_regex(regex="=([^=]+)$", string=Stdout).g1 AS Value
+            FROM execve(argv=[format(format= 'printenv')], sep="\n")
```

Comment:
this just shells out to  printenv - can you plain the advantage over the built in environ() plugin?

https://docs.velociraptor.app/vql_reference/plugin/environ/

Reply:
Sorry, I did not know about this plugin. environ() plugin display all the env variables?

Reply:
Yes it does.